pkg/apis/azureprovider/v1alpha1/azureclusterproviderconfig_types.go

```diff
@@ -32,15 +32,33 @@ type AzureClusterProviderSpec struct {
 	ResourceGroup string `json:"resourceGroup"`
 	Location      string `json:"location"`
 
-	// CACertificate is a PEM encoded CA Certificate for the control plane nodes.
-	CACertificate []byte `json:"caCertificate,omitempty"`
+	// CAKeyPair is the key pair for ca certs.
+	CAKeyPair KeyPair `json:"caKeyPair,omitempty"`
 
-	// CAPrivateKey is a PEM encoded PKCS1 CA PrivateKey for the control plane nodes.
-	CAPrivateKey []byte `json:"caKey,omitempty"`
+	//EtcdCAKeyPair is the key pair for etcd.
+	EtcdCAKeyPair KeyPair `json:"etcdCAKeyPair,omitempty"`
+
+	// FrontProxyCAKeyPair is the key pair for FrontProxyKeyPair.
+	FrontProxyCAKeyPair KeyPair `json:"frontProxyCAKeyPair,omitempty"`
+
+	// SAKeyPair is the service account key pair.
+	SAKeyPair KeyPair `json:"saKeyPair,omitempty"`
```

Comment:
I'm wondering if it would make sense to publish these external to the AWS/Azure providers and import them into both?

Reply:
Maybe add them to cluster-api itself, rather than in the provider? Makes sense, since certs are provider agnostic as far as I'm aware.

Reply:
Long term, yes. Shorter term, we can stage in a separate repo.

Reply:
Agreed that we should eventually pull the reusable pieces into capi proper.